### PR TITLE
Upgrade build configuration

### LINKED ISSUE #11: webpack-cli needs to be updated
I tried doing a build and got an error:

```
me@here:~/git/snowplow-js-analytics-sdk$ npm run build

> snowplow-analytics-sdk@0.1.0 build /home/me/git/snowplow-js-analytics-sdk
> webpack --config=webpack.config.js

/home/me/git/snowplow-js-analytics-sdk/node_modules/webpack-cli/bin/config-yargs.js:89
				describe: optionsSchema.definitions.output.properties.path.description,
				                                           ^

TypeError: Cannot read property 'properties' of undefined
    at module.exports (/home/me/git/snowplow-js-analytics-sdk/node_modules/webpack-cli/bin/config-yargs.js:89:48)
    at /home/me/git/snowplow-js-analytics-sdk/node_modules/webpack-cli/bin/webpack.js:60:27
    at Object.<anonymous> (/home/me/git/snowplow-js-analytics-sdk/node_modules/webpack-cli/bin/webpack.js:515:3)
```

There are reports of this being caused by webpack 4.20.0, e.g. https://github.com/webpack/webpack/issues/8082

The fix is to use webpack-cli 3.1.1.  Making that change fixed it for me.

### PR DESCRIPTION
**Pull request by dokmic**
This pull-request fixes the following:
- Add ECMAScript Modules support.
- Upgrade dependencies.
- Migrate to the automated builds.
- Migrate to eslint.
- Migrate to prettier.
- Migrate to rollup.
- Migrate to the Airbnb style guide.

Closes #11 .

### CODE REVIEW

**Comment by chuwy** [comment]
Hey @dokmic, sorry for the delay - I'll review it today!

**Comment by dokmic** [comment]
Hey @chuwy, did you have a chance to take a look at the change?

File: src/event.ts [hunk 1]
```diff
@@ -67,13 +68,13 @@ export interface Event {
   mkt_term: string;
   mkt_content: string;
   mkt_campaign: string;
-  contexts: any;
+  contexts: unknown;
```
Comment (paulboocock) [inline src/event.ts]: Switching to unknown is a nice idea, but this is a backwards incompatible change. I feel this change pushes this beyond a PATCH release and is at least a MINOR, I appreciate we are still on MAJOR v0 so I think this is fine getting a MINOR bump to 0.2.0.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "snowplow-analytics-sdk",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Comment (paulboocock) [inline package.json]: 0.2.0? Feels like a big change for a patch (plus changes to the Event interface).

File: src/index.ts [hunk 3]
```diff
@@ -10,60 +10,57 @@
...
-    .reduce((output: Field[], key, index) => {
-      if (event[index]) {
-        try {
-          output.push(...EVENT_STRUCTURE[key](key, event[index]));
-        } catch (e) {
-          errors.push(e.message);
+  const errors = [] as string[];
+  const output = Object.keys(EVENT_STRUCTURE)
+    .reduce(
+      (output, key, index) => {
```
Comment (paulboocock) [inline src/index.ts]: This line fails linting locally for me due to the new line after `.reduce(` and the new line at the bottom of the function before `[] as Field[],`. Running --fix reshuffles this section a little and fixes linting locally.
I'm not entirely sure why this passes on Travis and fails locally however, I have tested the altered version on Travis and it also passes: https://travis-ci.org/snowplow-incubator/snowplow-js-analytics-sdk/builds/617177080

File: package.json (hunk 2 above)
Comment (goodits) [reply package.json]: Is this a symptom that it should be a 1.x.x repo already?

**Comment by dokmic** [comment]
Hey @chuwy and @paulboocock,

Thank you for the review. I've fixed what you discovered:
- Updated the `CHANGELOG` to be upside down;
- Did a minor bump (`0.2.0` now);
- Fixed formatting causing failed linting.

**Pull request merged**